File: backend/express/payments-ms/src/factory/response.factory.ts
// src/factory/response.factory.ts
import { Response } from "express";
import { ApiResponse } from "../dtos/api-response.dto";

export class ResponseFactory {
  static created<T>(res: Response, message: string, data: T) {
    return res.status(201).json(ApiResponse.success(message, data));
  }

  static ok<T>(res: Response, message: string, data: T) {
    return res.status(200).json(ApiResponse.success(message, data));
  }

  static error<T>(res: Response, statusCode: number, message: string, data?: T) {
    return res.status(statusCode).json(ApiResponse.error(message, data ?? null));
  }
}

File: backend/webflux/notifications-ms/src/main/java/com/example/notifications/factory/ResponseFactory.java
package com.example.notifications.factory;

import com.example.notifications.dto.ApiResponse;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class ResponseFactory {

    public static <T> ResponseEntity<ApiResponse<T>> created(String message, T data) {
        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(ApiResponse.success(message, data));
    }

    public static <T> ResponseEntity<ApiResponse<T>> ok(String message, T data) {
        return ResponseEntity
                .status(HttpStatus.OK)
                .body(ApiResponse.success(message, data));
    }

    public static <T> ResponseEntity<ApiResponse<T>> error(HttpStatus status, String message, T data) {
        return ResponseEntity
                .status(status)
                .body(ApiResponse.error(message, data));
    }
}
